cmake/Install.cmake

```diff
@@ -121,4 +134,21 @@ function(ly_install_files)
         COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME} # use the default for the time being
     )
 
-endfunction()
\ No newline at end of file
+endfunction()
+
+#! ly_install_run_code: specifies code to be added to the install process (will run at install time)
+#
+# \notes: 
+#  - refer to cmake's install(CODE documentation for more information
+#
+function(ly_install_run_code CODE)
+
+    if(NOT LY_INSTALL_ENABLED)
+        return()
+    endif()
+
+    install(CODE ${CODE}
+        COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME} # use the default for the time being
+    )
```

Comment:
Explicitly specifying the default component here isn't completely necessary unless you want to expose modifying the component at the function scope.

Reply:
talked to Scott offline, `CMAKE_INSTALL_DEFAULT_COMPONENT_NAME` is the default assumed component in every `install` call, so there is no need to override it if is always the default.
I am leaving it anyway since we will start categorizing at some point and this will make it easier to find where to adjust. (unless someone feels strong about not having it, I can still locate all the calls to `install`)